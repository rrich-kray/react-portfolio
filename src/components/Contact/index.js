import React, { useState } from "react";
import { validateEmail } from "../../utils/helpers";

const Contact = () => {
  // when input content is changed, handleInputChange is invoked, which
  const [username, setUsername] = useState("");
  const [email, setEmail] = useState("");
  const [message, setMessage] = useState("");
  const [errorMessage, setErrorMessage] = useState("");

  const handleInputChange = (e) => {
    if (e.target.name === "email") setEmail(e.target.value);
    if (e.target.name === "username") setUsername(e.target.value);
    if (e.target.name === "message") setMessage(e.target.value);
  };

  const handleFormSubmit = (e) => {
    e.preventDefault();
    if (!email || !username || !message || !validateEmail(email)) {
      setErrorMessage("Invalid input detected");
      return;
    }

    setUsername("");
    setEmail("");
    setMessage("");
  };
  return (
    <div className="contact d-flex flex-column justify-content-center align-items-center">
      <form className="form d-flex flex-column justify-content-center align-items-center">
        <label for="username">Username:</label>
        <input
          type="text"
          name="username"
          onChange={handleInputChange}
          value={username}
        />
        <label for="email">Email</label>
        <input
          type="text"
          name="email"
          onChange={handleInputChange}
          value={email}
        />
        <label for="message">Message</label>
        <input
          type="text"
          name="message"
          onChange={handleInputChange}
          value={message}
        />
      </form>
      <button
        className="submit-btn nav-btn"
        onClick={handleFormSubmit}
        onChange={handleInputChange}
      >
        Submit Message
      </button>
      {errorMessage && (
        <div>
          <p className="error-text">{errorMessage}</p>
        </div>
      )}
    </div>
  );
};

export default Contact;
