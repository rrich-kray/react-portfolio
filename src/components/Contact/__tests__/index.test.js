import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Contact from "../index";

afterEach(cleanup);

describe("Component renders", () => {
  it("renders", () => {
    render(<Contact />);
  });

  it("Component matches DOM structure", () => {
    const { asFragment } = render(<Contact />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("Contact component renders", () => {
    const { getByTestId } = render(<Contact />);
    expect(getByTestId("contact").toHaveTextContent("Contact"));
  });

  it("email renders", () => {
    const { getByTestId } = render(<Contact />);
    expect(getByTestId("email")).toHaveTextContent(
      "Email: rrich.kray.93@gmail.com"
    );
  });

  it("GitHub renders", () => {
    const { getByTestId } = render(<Contact />);
    expect(getByTestId("gh")).toHaveTextContent(
      "GitHub: https://rrich-kray.github.io"
    );
  });
});
