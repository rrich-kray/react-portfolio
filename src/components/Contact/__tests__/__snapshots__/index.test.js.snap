// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component renders Component matches DOM structure 1`] = `
<DocumentFragment>
  <div
    class="contact"
  >
    <div
      class="panel-container d-flex flex-column justify-content-center align-items-center"
    >
      <div
        class="left-panel d-flex justify-content-center align-items-center w-50"
      >
        <h1>
          Contact
        </h1>
      </div>
      <div
        class="right-panel d-flex flex-column justify-content-center align-items-center w-50"
      >
        <h5>
          Email: rrich.kray.93@gmail.com
        </h5>
        <h5>
          GitHub: https://rrich-kray.github.io
        </h5>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
